SportsHub/SportsHub.Application/Services/UserService.cs

```diff
@@ -12,6 +13,11 @@ public UserService(IUnitOfWork unitOfWork)
             this.unitOfWork = unitOfWork;
         }
 
+        public async Task<User?> GetByEmailOrUsernameAsync(string input)
+        {
+            return  await unitOfWork.UserRepository.GetByUsernameAsync(input) ?? 
+                await unitOfWork.UserRepository.GetByEmailAsync(input);
```

Comment:
it still does two requests to the database if there is no User with UserName.

Reply:
It should be fixed now